go/pkg/cache/singleflightcache/singleflightcache.go

```diff
@@ -5,86 +5,42 @@
 package singleflightcache
 
 import (
-	"fmt"
 	"sync"
 )
 
 // Cache is a cache that supports single-flight value computation.
 type Cache struct {
-	// store is the actual cache where data is stored.
 	store sync.Map
-	// locks is a map of locks per key. This map is used to ensure only one reader/writer of
-	// the key can have write access.
-	locks sync.Map
-	// mu is a mutex to ensure exclusive access to the locks map in the case of Delete.
-	mu sync.RWMutex
+}
+
+type entry struct {
+	compute sync.Once
```

Comment:
Is it possible to use https://pkg.go.dev/golang.org/x/sync/singleflight in Cache instead of having sync.Once here?

Reply:
https://github.com/golang/sync/blob/036812b2e83c/singleflight/singleflight.go#L46-L71 seems overkill?

Reply:
ack

Reply:
Also with `singleflight` we would need to ensure we don't call the `compute` function if the value is loaded from the cache, since `singleflight` only dedups in flight computations and not all computations.